app/scripts/directives/fromFileDialog.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+angular.module('openshiftConsole').component('fromFileDialog', {
+  controller: [
+    '$scope',
+    '$timeout',
+    'DataService',
+    FromFileDialog
+  ],
+  controllerAs: '$ctrl',
+  bindings: {
+    visible: '<',
+    project: '<', //handle create project optionally
+    context: '<',
+    onDialogClosed: '&'
+  },
+  templateUrl: 'views/directives/from-file-dialog.html'
+});
+
+function FromFileDialog($scope, $timeout, DataService) {
+  var ctrl = this;
+
+  ctrl.$onInit = function() {
+    ctrl.alerts = {};
+    ctrl.loginBaseUrl = DataService.openshiftAPIBaseUrl();
+  };
+
+  function getIconClass() {
+    var icon = _.get(ctrl, 'template.metadata.annotations.iconClass', 'fa fa-cubes');
```

Comment:
Wondering if we should use the `iconClass` filter, which handles template icons. But it defaults to `fa-clone` for templates. This makes me wonder if we shouldn't use `fa-clone` as our default in the catalog.

I'm OK with this as-is, but we might open an issue to decide what to do since we can show different default icons for the new experience vs. old.

https://github.com/openshift/origin-web-console/blob/master/app/scripts/filters/resources.js#L53